### PR TITLE
feat: Add ESLINT_D_MAX_INSTANCES env variable. (#201) 

### LINKED ISSUE #201: Feature Request: Configurable max instances
I would like to be able to reduce the number of instances eslint_d keeps in memory.  Currently eslint_d is using 1.3gb of memory with all 10 instances cached.  Rather than eslint_d restarting regularly it'd be nice to be able to reduce this to 3-4 instances.

A way of configuring this could be via environment variables
```
export ESLINT_D_MAX_INSTANCES=5
eslint_d ./file.js
```

### PR DESCRIPTION
**Pull request by connorgmeehan**
Hey I left this branch on my local machine and figured I'd just push it.  This resolves #201 by adding an environment variable to configure the max instances so it can be reduced/increased from 10.  
It just needs some help getting the error logging working.  I can't print to std out. Would you be able to help me with this?

### CODE REVIEW

File: lib/caches.js [hunk 1]
```diff
@@ -6,7 +6,25 @@ const resolver = require('./resolver');
...
 
-const lru_cache = new LRU(10);
+// Setup LRU cache + error checking user config.
+// 10 by default unless ESLINT_D_MAX_INSTANCES env variable is set.
+const { ESLINT_D_MAX_INSTANCES } = process.env;
+let maxInstances = ESLINT_D_MAX_INSTANCES
+  ? Number.parseInt(ESLINT_D_MAX_INSTANCES, 10)
+  : 10;
+
+if (Number.isInteger(maxInstances)) {
```
Comment (watson28) [inline lib/caches.js:16]:
```suggestion
if (!Number.isInteger(maxInstances)) {
```

**Comment by uloco** [comment]
Really looking forward to this. Any blocker that can be removed here?

**Comment by mantoni** [comment]
@uloco Thanks for offering help.

This will need test cases and the validation should happen when starting the server so that we can log to stderr or fail. Currently this happens inside of the daemon and the error isn't shown.

**Comment by uloco** [comment]
I'll see what I can do thx

**Comment by mantoni** [comment]
This is obsolete with the new major `v14.0.0` (see #294). The new release spawns a separate instance per eslint installation. Please see the README on how to control the lifetime of each instance.

**Pull request closed without merging**